### ISSUE TITLE
Failing to bootstrap control plane with v0.1.0-alpha.3

### ISSUE THREAD

**Issue by zioc** [rancher-sandbox/cluster-api-provider-rke2#146]
**What happened:**

While attempting to upgrade from v0.1.0-alpha.2 to v0.1.0-alpha.3 (without changing anything else) first control plane node provisionning is stuck with controlplane controller looping on following messages:

```
I0612 15:47:37.061255       1 rke2controlplane_controller.go:499]  "msg"="Initializing control plane" "Desired"=3 "Existing"=0 "RKE2ControlPlane"={"name":"management-cluster-control-plane","namespace":"default"} "controller"="rke2controlplane" "controllerGroup"="controlplane.cluster.x-k8s.io" "controllerKind"="RKE2ControlPlane" "name"="management-cluster-control-plane" "namespace"="default" "reconcileID"="7f678957-46e5-40be-9149-2d8ef42db0d7"
E0612 15:47:37.085811       1 controller.go:329]  "msg"="Reconciler error" "error"="control plane has already been initialized, found 1 owned machine for cluster default/management-cluster: controller cache or management cluster is misbehaving" "RKE2ControlPlane"={"name":"management-cluster-control-plane","namespace":"default"} "controller"="rke2controlplane" "controllerGroup"="controlplane.cluster.x-k8s.io" "controllerKind"="RKE2ControlPlane" "name"="management-cluster-control-plane" "namespace"="default" "reconcileID"="7f678957-46e5-40be-9149-2d8ef42db0d7"
```

After some investigation it turns out to be caused by the fact that cluster fails to [retrieve control plane nodes](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/main/controlplane/internal/controllers/rke2controlplane_controller.go#LL428C10-L428C10).

Control plane nodes are filtered out, as they are labelled with `cluster.x-k8s.io/control-plane-name` instead of `cluster.x-k8s.io/control-plane`. We should use `clusterv1.MachineControlPlaneLabel` instead of `clusterv1.MachineControlPlaneNameLabel` [here](https://github.com/rancher-sandbox/cluster-api-provider-rke2/blob/main/pkg/rke2/control_plane.go#LL223C3-L223C41), as this the the label that is used to[ filter control plane machines](https://github.com/kubernetes-sigs/cluster-api/blob/main/util/collections/machine_filters.go#L224)

**Comment by furkatgofurov7** [comment]
@zioc thanks for the report. 

I think your analysis is pretty much accurate to me. Since you have done all the investigation work, would you like to also send a PR to fix it?

**Comment by richardcase** [comment]
This has been fixed as part of #136

**Comment by furkatgofurov7** [comment]
> @richardcase I don't see a linked PR since #136 is the issue itself and still open. Did you mean this issue will be taken care by the fix for #136?

based on the slack discussion you meant the same, nvm
